Sources/Network/NetworkAuthenticator.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Authenticator.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 03/07/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public protocol NetworkAuthenticator {
+    func authenticate(request: URLRequest, _ completion: @escaping (URLRequest) -> Void)
```

Comment:
1. about the naming, for me it seems ok, i'm calling the function to `authenticate` the request and then it returns the completion when it's ready, so it seems like a completion closure, however shout your name suggestion if you have one.
2. When you mean requests, are those network requests? Not necessarily, they can be something that takes more time.